polkadot/runtime/parachains/src/paras_inherent/tests.rs

```diff
@@ -2127,19 +2235,44 @@ mod sanitizers {
 
 				// Verify the initial state is as expected
 				assert_eq!(
-					all_backed_candidates_with_core.get(0).unwrap().0.validity_votes().len(),
+					all_backed_candidates_with_core
+						.get(&ParaId::from(1))
+						.unwrap()
+						.iter()
+						.next()
+						.unwrap()
+						.0
+						.validity_votes()
+						.len(),
 					2
 				);
-				let untouched = all_backed_candidates_with_core.get(1).unwrap().0.clone();
+				let untouched = all_backed_candidates_with_core
+					.get(&ParaId::from(2))
+					.unwrap()
+					.iter()
+					.next()
+					.unwrap()
+					.0
+					.clone();
 
-				assert!(filter_backed_statements_from_disabled_validators::<Test>(
+				filter_backed_statements_from_disabled_validators::<Test>(
 					&mut all_backed_candidates_with_core,
 					&<shared::Pallet<Test>>::allowed_relay_parents(),
-					core_index_enabled
-				));
+					core_index_enabled,
+				);
 
 				assert_eq!(all_backed_candidates_with_core.len(), 1);
-				assert_eq!(all_backed_candidates_with_core.get(0).unwrap().0, untouched);
+				assert_eq!(
+					all_backed_candidates_with_core
+						.get(&ParaId::from(2))
+						.unwrap()
+						.iter()
+						.next()
+						.unwrap()
+						.0,
+					untouched
+				);
+				assert_eq!(all_backed_candidates_with_core.get(&ParaId::from(1)), None);
```

Comment:
We are still missing some actual elastic scaling tests here, right?

Reply:
these should be much easier to implement with benchbuilder elastic scaling support like https://github.com/paritytech/polkadot-sdk/pull/3678

Reply:
yes, I'm adding more tests here as we speak. Will also make use of the benchbuilder for some tests

Reply:
done, added more tests